### ISSUE TITLE
(Thread 1) Malformed message from xxx.xxx.xxx.xxx

### ISSUE THREAD

**Issue by corkapple** [SadBatman/cryptonote-sumokoin-pool#14]
2017-12-19 12:54:09 [pool] (Thread 1) Malformed message from 157.xxx.39.28: gcZ9
                                                                               �����n���hӳ�����1�f1N��& ��5/�
2017-12-19 12:54:09 [pool] (Thread 4) Malformed message from 157.xxx.39.28: ��Z9
|~E�u��6�F���                                                                  ��i�(Y
2▮17↑12↑19 12:54:1▮ [⎻⎺⎺┌] (T▒⎼e▒d 4) M▒┌°⎺⎼└ed └e⎽⎽▒±e °⎼⎺└ 157↓xxx↓39↓28: �Z9
 
when this happens hash rate drops down. I use byobu for screen sessions.

installed as per instructions in your readme file.

**Comment by SadBatman** [comment]
Interesting.  Does this happen often ? This could be the result of a port scan or a misconfigured miner.

Do you have the raw logs ?

**Comment by corkapple** [comment]
[admin_log.zip](https://github.com/SadBatman/cryptonote-sumokoin-pool/files/1571842/admin_log.zip)

**Comment by corkapple** [comment]
I actually used iptables to block few of those ip's and it was ok for a moment, now back. Could be port scan as you said or wrongly configured miner client

**Comment by SadBatman** [comment]
Yeah but nevertheless it should not mess up your hashrate so I'm glad you
report this.

We do have IP banning but I'm not sure this case is handled by it. I'll
check the code later (currently commuting)
